### PR TITLE
[issue-426] add LifecycleScopedRelationship and SoftwareDependencyRelationship

### PR DESCRIPTION
**Pull request by armintaenzertng**
part of #426

### CODE REVIEW

**Review by meretp — APPROVED** [review]
LGTM, just somem inor comments that are not merge-blocking. But as discussed earlier I think we should wait for the spdx-3-model PR to be merged.

File: src/spdx_tools/spdx3/model/relationship.py [hunk 1]
```diff
@@ -54,8 +54,8 @@ class RelationshipType(Enum):
 
 class RelationshipCompleteness(Enum):
     INCOMPLETE = auto()
-    KNOWN = auto()
-    UNKNOWN = auto()
+    COMPLETE = auto()
+    NOASSERTION = auto()
```
[the remark this answers is not in the document]
Comment (armintaenzertng) [reply src/spdx_tools/spdx3/model/relationship.py:58]: It actually is part of the PR: https://github.com/spdx/spdx-3-model/pull/69/files#diff-fa3471f59f9fd3a5a28e841955d9f4d028701e05c6ca4af929f753ee944d3177

File: tests/spdx3/model/test_lifecycle_scoped_relationship.py
[the remark this answers is not in the document]
Comment (armintaenzertng) [reply tests/spdx3/model/test_lifecycle_scoped_relationship.py:1]: added

File: tests/spdx3/model/software/test_software_dependency_relationship.py
[the remark this answers is not in the document]
Comment (armintaenzertng) [reply tests/spdx3/model/software/test_software_dependency_relationship.py:1]: added

File: src/spdx_tools/spdx3/model/relationship.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (meretp) [reply src/spdx_tools/spdx3/model/relationship.py:58]: :scream: must have overseen it, nevermind

File: tests/spdx3/model/test_lifecycle_scoped_relationship.py
[the remark this answers is not in the document]
Comment (meretp) [reply tests/spdx3/model/test_lifecycle_scoped_relationship.py:1]: thanks

**Review by meretp — APPROVED** [review]
Thanks for adding the writer, I think this ones complete now, hoepfully the PR in the spdx-3-model will be merged soon.

**Pull request merged**